@messageformat/runtime/messages import not working in TypeScript

When trying to import `@messageformat/runtime/messages` in TypeScript, it can't find typings for that submodule. TypeScript build fails with error `TS7016: Could not find a declaration file for module '@messageformat/runtime/messages'`.

#### Steps to reproduce
- create a TypeScript project with `"strict": true` or `"noImplicitAny": true`
- try to import anything from `@messageformat/runtime/messages` in a `.ts` file
- run `npx tsc`

#### Expected result
- no error

#### Actual result
- error thrown:
  ```markdown
  file.ts:1:20 - error TS7016: Could not find a declaration file for module '@messageformat/runtime/messages'. '/src/xxx/node_modules/@messageformat/runtime/messages.js' implicitly has an 'any' type.
    Try `npm i --save-dev @types/messageformat__runtime` if it exists or add a new declaration (.d.ts) file containing `declare module '@messageformat/runtime/messages';`

  1 import * as M from '@messageformat/runtime/messages';
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ```

#### Reproduction repository
[katemihalikova/repro--messageformat-runtime-messages](https://github.com/katemihalikova/repro--messageformat-runtime-messages).

#### Workaround
Import from `@messageformat/runtime/lib/messages` instead - but this is not defined in Webpack `exports` field in `package.json` so it should be considered a private API.

For a fix, I was thinking about adding a `messages.d.ts` into root of the package, alongside root `messages.js`, but the js file exports the original default export as a node export instead. This would require a legacy typings of `export = Messages;` that is quite cumbersome as every export needs to be renamed and TypeScript requires `"esModuleInterop": true` flag to work.

```typescript
import M, { MessageData as MD, MessageFunction as MF } from './lib/messages';

declare const Messages: typeof M;
declare namespace Messages {
  export type MessageData = MD;
  export type MessageFunction = MF;
}

export = Messages;
```

---

To sum it up, is it ok to just import from `@messageformat/runtime/lib/messages`? Or is some kind of change needed to support plain TypeScript (e.g. in intellisense)?

--- a/packages/runtime/messages.d.ts
+++ b/packages/runtime/messages.d.ts
@@ -1,0 +1,2 @@
+export * from './lib/messages';
+export { default } from './lib/messages';

--- a/packages/runtime/package.json
+++ b/packages/runtime/package.json
@@ -21,7 +21,7 @@
   "files": [
     "esm/",
     "lib/",
-    "messages.js"
+    "messages.*"
   ],
   "type": "commonjs",
   "main": "./lib/runtime.js",

